sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/CreateTables.java

```diff
@@ -78,20 +80,38 @@ CreateTables withTestServices(BigQueryServices bqServices) {
   @Override
   public PCollection<KV<TableDestination, TableRow>> expand(
       PCollection<KV<TableDestination, TableRow>> input) {
+    List<PCollectionView<Map<String, String>>> sideInputs = Lists.newArrayList();
+    if (schemaFunction.getSideInput() != null) {
+      sideInputs.add(schemaFunction.getSideInput());
+    }
     return input.apply(
         ParDo.of(
             new DoFn<KV<TableDestination, TableRow>, KV<TableDestination, TableRow>>() {
+              SchemaFunction schemaFunctionCopy;
+
+              @StartBundle
+              public void startBundle(Context c) {
+                this.schemaFunctionCopy = SerializableUtils.clone(schemaFunction);
+              }
+
               @ProcessElement
               public void processElement(ProcessContext context)
                   throws InterruptedException, IOException {
+                if (schemaFunctionCopy.getSideInput() != null) {
```

Comment:
I'm not sure I understand. We access the side input separately on each key (setSideInputValue), so the side input should match the current window.